src/core/globalstate.ts

```diff
@@ -143,6 +143,7 @@ export class MobXGlobals {
 
 let canMergeGlobalState = true
 let isolateCalled = false
+const mockGlobal = {}
```

Comment:
Why did you move it here? Let's keep it closer to the actual use.

Reply:
There's a good reason to move it. I described the issue in the commit message. Currently when `getGlobal` is first invoked, it throws exception in case `mockGlobal` is returned because `mockGlobal` is not initialized yet.

Reply:
I can move the whole function upward in order to keep them closer if you prefer

Reply:
Just to make it clearer. This is in general the current code:

```js
export let globalState = (function() {
    const global = getGlobal()
    ... global._mobxGlobals ...
})()

const mockGlobal = {}
export function getGlobal() {
  ...
  return mockGlobal
}
```

since `globalState` is a IIFE which is invoked before the line `const mockGlobal = {}` is reached, the `mockGlobal` variable is still not initialized  with the `{}` value, therefore the call to `getGlobal` inside the IIFE will not work as expected.

Reply:
Ah, you are right, missed it's being called in IIFE. Yea, I think moving `getGlobal` on top as well makes sense to keep it collocated.

Reply:
done